### PR TITLE
[FIX] livechat : apply multiples fixes

### PR DESCRIPTION
**Pull request by dbeguin**
This PR fixes multiples bugs and inconsistent behaviours in livechat:

From now,
- livechat does not close after rating.
- a placeholder with email example is displayed in form 
  to receive the copy of the livechat conversation
- use visitor name for livechat session naming when visitor start livechat
- use main rating image instead of white one for livechat frontend view
- rating_last_image is now readonly
- rating per user in frontend view is now correct

See sub commits for more details

Task ID: 2076656

### CODE REVIEW

File: addons/im_livechat/static/src/js/im_livechat.js [hunk 1]
```diff
@@ -449,22 +449,21 @@ var Feedback = Widget.extend({
...
      * @param {Object} options
      */
-    _sendFeedback: function (options) {
+    _sendFeedback: function (reason) {
         var self = this;
         var args = {
             uuid: this._livechat.getUUID(),
             rate: this.rating,
-            reason : options.reason
+            reason : reason
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/js/im_livechat.js]: simply `reason,` is enough:

```js
var args = {
   // ...
  reason,
}
```

File: addons/im_livechat/static/src/js/im_livechat.js [hunk 2]
```diff
@@ -449,22 +449,21 @@ var Feedback = Widget.extend({
...
         };
         this.dp.add(session.rpc('/im_livechat/feedback', args)).then(function () {
-            if (options.close) {
+            if (reason) {
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/js/im_livechat.js]: `else` is inline with `}`:

```js
if (/* condition */) {
   // ...
} else {
   // ...
}
```

File: addons/im_livechat/static/src/js/im_livechat.js [hunk 3]
```diff
@@ -492,9 +491,10 @@ var Feedback = Widget.extend({
      * @private
      */
     _onClickSend: function () {
+        this.$('.o_livechat_rating_reason').hide();
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/js/im_livechat.js]: `.remove()`

File: addons/im_livechat/static/src/js/im_livechat.js (hunk 1 above)
Comment (dbeguin) [reply addons/im_livechat/static/src/js/im_livechat.js]: ??

Comment (dbeguin) [reply addons/im_livechat/static/src/js/im_livechat.js]: ha yes, `reason` and not `reason: reason`

Comment (alexkuhn) [reply addons/im_livechat/static/src/js/im_livechat.js]: yup :)

File: addons/im_livechat/static/src/js/im_livechat.js [hunk 4]
```diff
@@ -449,22 +449,21 @@ var Feedback = Widget.extend({
      * @private
      * @param {Object} options
      */
-    _sendFeedback: function (options) {
+    _sendFeedback: function (reason) {
```
[the remark this answers is not in the document]
Comment (dbeguin) [reply addons/im_livechat/static/src/js/im_livechat.js]: within same file, I'll keep same way to declare function.

**Comment by dbeguin** [comment]
@alexkuhn : noooo it was broken :D

**Comment by alexkuhn** [comment]
really? did the changes locally and it worked fine though... 😕

**Comment by dbeguin** [comment]
@alexkuhn : it was sending twice the rating and never the reason. Weird, because I tested before push first time and it worked.. beh !

**Comment by alexkuhn** [comment]
Oops, I forgot something in the review: the suggested `.remove()` has to be done after retrieving the value from the `textarea`:
https://github.com/odoo/odoo/pull/37511/files#diff-8576c3a6f81796ad0202d607ce0352ceR495
As for the rating sent twice, it happens if you click on send then on the smiley, but you just fixed that :p

File: addons/im_livechat/controllers/main.py [hunk 5]
```diff
@@ -87,13 +87,17 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, **k
...
+            visitor_sudo = request.env['website.visitor']._get_visitor_from_request()
+            if visitor_sudo:
+                anonymous_name = visitor_sudo.display_name
+            # if geoip, add the country name to the anonymous name
+            if request.session.geoip:
+                # get the country of the anonymous person, if any
+                country_code = request.session.geoip.get('country_code', "")
+                country = request.env['res.country'].sudo().search([('code', '=', country_code)], limit=1) if country_code else None
+                if country:
+                    anonymous_name, country_id = _("%s (%s)") % (anonymous_name, country.name), country.id
```
Comment (dbeguin) [inline addons/im_livechat/controllers/main.py]: don't translate David Beguin

File: addons/im_livechat/controllers/main.py [hunk 6]
```diff
@@ -87,13 +87,17 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, **k
...
+            visitor_sudo = request.env['website.visitor']._get_visitor_from_request()
+            if visitor_sudo:
+                anonymous_name = visitor_sudo.display_name
+            # if geoip, add the country name to the anonymous name
+            if request.session.geoip:
+                # get the country of the anonymous person, if any
+                country_code = request.session.geoip.get('country_code', "")
+                country = request.env['res.country'].sudo().search([('code', '=', country_code)], limit=1) if country_code else None
+                if country:
+                    anonymous_name, country_id = "%s (%s)" % (anonymous_name, _(country.name)), country.id
```
Comment (tde-banana-odoo) [inline addons/im_livechat/controllers/main.py]: Don't manually translate country.name
You may also declare the two variables in two lines to ease understanding

Comment (tde-banana-odoo) [reply addons/im_livechat/controllers/main.py]: Define a "complete_name" or something similar that is either anonymous_name, visitor_sudo.display_name + country thingy. That way it is easier to understand.

File: addons/im_livechat/controllers/main.py [hunk 7]
```diff
@@ -87,13 +87,17 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, **k
...
+            visitor_sudo = request.env['website.visitor']._get_visitor_from_request()
+            if visitor_sudo:
+                anonymous_name = visitor_sudo.display_name
+            # if geoip, add the country name to the anonymous name
+            if request.session.geoip:
+                # get the country of the anonymous person, if any
+                country_code = request.session.geoip.get('country_code', "")
+                country = request.env['res.country'].sudo().search([('code', '=', country_code)], limit=1) if country_code else None
+                if country:
+                    anonymous_name, country_id = "%s (%s)" % (anonymous_name, country.name), country.id
```
Comment (tde-banana-odoo) [inline addons/im_livechat/controllers/main.py]: Don't manually translate country.name
You may also declare the two variables in two lines to ease understanding

Define a "complete_name" or something similar that is either anonymous_name, visitor_sudo.display_name + country thingy. That way it is easier to understand.

File: odoo/addons/base/models/ir_qweb_fields.py [hunk 8]
```diff
@@ -135,8 +135,8 @@ def user_lang(self):
 
         :returns: Model[res.lang]
         """
-        lang_code = self._context.get('lang') or 'en_US'
-        return self.env['res.lang']._lang_get(lang_code)
+        lang_code = self._context.get('lang')
+        return self.env['res.lang']._lang_get(lang_code) or self.env['res.lang'].search([('code', '=', 'en_US')])
```
Comment (tde-banana-odoo) [inline odoo/addons/base/models/ir_qweb_fields.py]: I would move this in a separate PR to be reviewed when we have some time to think about it.

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff
